src/server/dflycmd.cc

```diff
@@ -307,6 +302,27 @@ void DflyCmd::Flow(CmdArgList args, RedisReplyBuilder* rb, ConnectionContext* cn
   }
 #endif
 
+    std::optional<Replica::LastMasterSyncData> data = sf_->GetLastMasterData();
+    if (last_master_id.has_value() && data.has_value()) {
```

Comment:
```suggestion
    if (last_master_id && data) {
```

Reply:
`std::optional` is implicitly convertible to bool. See https://en.cppreference.com/w/cpp/utility/optional

Reply:
yes I think this is a matter of style preference. Dont think we follow a specific approach in our code

Reply:
yep yep I agree I am not trying to enforce anything just commenting that there is that interface as well.